src/main/java/org/carlspring/maven/commons/util/ArtifactUtils.java

```diff
@@ -219,7 +219,21 @@ public static String convertArtifactToPath(Artifact artifact)
                     !artifact.getClassifier().equals("") &&
                     !artifact.getClassifier().equals("null") ?
                     "-" + artifact.getClassifier() : "";
-            path += artifact.getType() != null ? "." + artifact.getType() : "";
+                    
+            String type = null;
+            if (artifact.getType() != null)
+            {
+                if ("maven-plugin".equals(artifact.getType()))
```

Comment:
I don't agree with this change. The type can only be `jar`, not `.jar` and not `maven-plugin.jar`. This is not valid because the artifact's path will be incorrect.

Assume the following:

    GAV: org.carlspring.foo:bar-maven-plugin:1.2.3:maven-plugin

The path for the above should be:

    org/carlspring/foo/bar-maven-plugin/1.2.3/bar-maven-plugin-1.2.3.jar

With the proposed fix, it will be:

    org/carlspring/foo/bar-maven-plugin/1.2.3/bar-maven-plugin-1.2.3-maven-plugin.jar

With your fix, you're sort of, unintentionally, adding the `maven-plugin` bit as a classifier instead.

If you create a simple test case and run your code through it, you'll see what I mean.